online/src/main/scala/ai/chronon/online/FetcherBase.scala

```diff
@@ -288,29 +290,47 @@ class FetcherBase(kvStore: KVStore,
               val streamingResponsesOpt =
                 streamingRequestOpt.map(responsesMap.getOrElse(_, Success(Seq.empty)).getOrElse(Seq.empty))
               val queryTs = request.atMillis.getOrElse(System.currentTimeMillis())
-              try {
-                if (debug)
-                  logger.info(
-                    s"Constructing response for groupBy: ${groupByServingInfo.groupByOps.metaData.getName} " +
-                      s"for keys: ${request.keys}")
-                constructGroupByResponse(batchResponseTryAll,
-                                         streamingResponsesOpt,
-                                         groupByServingInfo,
-                                         queryTs,
-                                         startTimeMs,
-                                         multiGetMillis,
-                                         context,
-                                         totalResponseValueBytes)
-              } catch {
-                case ex: Exception =>
-                  // not all exceptions are due to stale schema, so we want to control how often we hit kv store
-                  getGroupByServingInfo.refresh(groupByServingInfo.groupByOps.metaData.name)
-                  context.incrementException(ex)
-                  ex.printStackTrace()
-                  throw ex
+              val groupByResponse: Map[String, AnyRef] =
+                try {
+                  if (debug)
+                    logger.info(
+                      s"Constructing response for groupBy: ${groupByServingInfo.groupByOps.metaData.getName} " +
+                        s"for keys: ${request.keys}")
+                  constructGroupByResponse(batchResponseTryAll,
+                                           streamingResponsesOpt,
+                                           groupByServingInfo,
+                                           queryTs,
+                                           startTimeMs,
+                                           multiGetMillis,
+                                           context,
+                                           totalResponseValueBytes)
+                } catch {
+                  case ex: Exception =>
+                    // not all exceptions are due to stale schema, so we want to control how often we hit kv store
+                    getGroupByServingInfo.refresh(groupByServingInfo.groupByOps.metaData.name)
+                    context.incrementException(ex)
+                    ex.printStackTrace()
+                    throw ex
+                }
+              if (groupByServingInfo.groupBy.hasDerivations) {
+                val derivedMapTry: Try[Map[String, AnyRef]] = Try{
```

Comment:
Trying to apply the new error handling logic we introduced in this PR: https://github.com/airbnb/chronon/pull/716